### ISSUE TITLE
Cannot arm or disarm location_home

### ISSUE THREAD

**Issue by brenmcg57** [twrecked/hass-aarlo#843]
I had a working installation with an automation to arm and disarm the Home location. This stopped working a few days ago. When I try clicking the Arm Home or Arm Away buttons of the location_home card, I get the following popup at the bottom of the screen:

`Failed to call service alarm_control_panel/alarm_arm_home. 'NoneType' object has no attribute 'get'`

I only have the default Aarlo recommended config sections.

I tried a clean HA install and got the same result. I also tried allowing upgrades to the .8 beta versions. No change in behavior.
Keep in mind that my 2FA is working fine. I can see the camera and interact with their cards.

Maybe not interesting, but now the user_location_home (not to be confused with location_home) card has both arm and disarm buttons and they are interactive, but my Arlo app shows no effect of clicking them. Only the location_home buttons did anything before.

**Comment by twrecked** [comment]
Anything in here help: https://github.com/twrecked/hass-aarlo/issues/687

Other than that I will need
- the version you are using
- your config (minus the sensitive pieces)
- and some debug logs, see [here](https://github.com/twrecked/hass-aarlo#very-verbose-debug) and [here](https://github.com/twrecked/hass-aarlo#hiding-sensitive-data) on how to encrypt the logs.

**Comment by anachronox99** [comment]
Hi I have the same problem. 

my config:

aarlo:
  host: https://my.arlo.com
  username: XXX
  password: XXX
  db_motion_time: 30
  db_ding_time: 10
  recent_time: 10
  last_format: '%d-%m %H:%M'
  refresh_devices_every: 2
#  verbose_debug: True
  tfa_host: 'imap.gmail.com'
  tfa_username: XXX
  tfa_password: XXX
  backend: sse

alarm_control_panel:
  - platform: manual  
    code: XXX
    code_arm_required: true
    name: Home alarm
    #pending_time: 3
    arming_time: 3
  - platform: aarlo
    away_mode_name: Armed
#    home_mode_name: home
    disarmed_mode_name: Disarmed
#    night_mode_name: night
    trigger_time: 30
    alarm_volume: 8

**Comment by brenmcg57** [comment]
Update: I don't think its the aarlo plugin right now. I think I might have run afoul of some Arlo WAF. I cannot login to my dashboard in the browser from my home network. The instant I click login I get a banner on their site that I have No Internet. (lol ok,) or an instant "Request Timed Out."

If I get behind another public IP, like my neighbor's wifi, I can authenticate. I came across this thread and I am trying to confirm I'm having a similar issue: https://community.arlo.com/t5/Arlo-Pro/You-are-unable-to-access-ocapi-app-arlo-com/m-p/1920672

**Comment by brenmcg57** [comment]
I was able to confirm that I am having the issue linked in my post just above this. Chrome devtools shows me:
`Access to XMLHttpRequest at 'https://ocapi-app.arlo.com/api/auth' from origin 'https://my.arlo.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.`
I believe I am dealing with two issues. I originally posted when I could still interact with my cameras but I could not change the alarm panel mode. Now I have a new issue, where I am blocked from authenticating to my arlo cloud instance from my public IP. I am trying to get Arlo support to unblock my IP. Maybe that'll magically fix my first issue.

**Comment by twrecked** [comment]
Good luck with the blocking issue. Is there any way you can get a new IP?

@anachronox99 (and @brenmcg57 when you're unblocked) try removing the `host:` config, getting locations from `my.arlo.com` fails.

**Comment by anachronox99** [comment]
HI @twrecked -> so remove the host line altogether, not replace it with some other values?

**Comment by twrecked** [comment]
Yes, remove it, the default (`https://myapi.arlo.com`) is correct.

**Comment by anachronox99** [comment]
OK did that and restarted. My Arlo app shows my status as "Arm Home" and home assistant shows status as armed_away. I then tried to disarm in HA from the Alarm panel lovelace element by clicking 'Disarm' and nothing happened. I then went to the Arlo app and changed status to 'Standby' and waited a minute - HA did not refresh it's status.

**Comment by twrecked** [comment]
@anachronox99  I'm going to need some logs.

Turn on [verbose debug](https://github.com/twrecked/hass-aarlo#very-verbose-debug).

And [encrypt it](https://github.com/twrecked/hass-aarlo#online) with the link in that section.

**Comment by anachronox99** [comment]
Here you go

[homeassistant.log](https://github.com/twrecked/hass-aarlo/files/13482671/homeassistant.log)
[homeassistant-1.log](https://github.com/twrecked/hass-aarlo/files/13482686/homeassistant-1.log)

**Comment by twrecked** [comment]
There was no arlo related debug in the logs. You need to update your config and reboot. There will be a lot of stuff from `aarlo` and `pyaarlo`.

**Comment by anachronox99** [comment]
[homeassistant.log](https://github.com/twrecked/hass-aarlo/files/13483287/homeassistant.log)
ok pls try this. I did a grep and there was some aarlo content not much though.

**Comment by twrecked** [comment]
My bad, I missed a step. You need to add this to you `configuration.yaml` to turn on the logging. You'll know when it's working - there is a lot of debug.

```yaml
logger:
  default: info
  logs:
    pyaarlo: debug
    custom_components.aarlo: debug
```

**Comment by brenmcg57** [comment]
OK it looks like my IP got unblocked after a cooldown period. I can now login again to the arlo dashboard from a browser. I want to jump back into helping with the diagnostics but I need to start over clean since I butchered up my HA instance. Since I'm starting from scratch, what installation method would you like me to try?

1. Stable
2. Stable upgraded to latest beta
3. Install beta from the start

**Comment by anachronox99** [comment]
> My bad, I missed a step. You need to add this to you `configuration.yaml` to turn on the logging. You'll know when it's working - there is a lot of debug.
> 
> ```yaml
> logger:
>   default: info
>   logs:
>     pyaarlo: debug
>     custom_components.aarlo: debug
> ```

New logs:
[homeassistant.log](https://github.com/twrecked/hass-aarlo/files/13495026/homeassistant.log)

**Comment by twrecked** [comment]
@brenmcg57 That's great. Sometimes you just need to give them time to forget you!

@anachronox99 Logs look good, I'll take a good look at them.

**Comment by anachronox99** [comment]
I made 3 attempts to arm the alarm. 

1. From the UI, I hit Disarm in the lovelace element. 

<img width="324" alt="image" src="https://github.com/twrecked/hass-aarlo/assets/16225843/5d8a2ec4-be54-424e-9883-5aad612f73d5">

2. From the dev panel, I called a service - first trying to arm it, then trying to disarm it. 

<img width="300" alt="image" src="https://github.com/twrecked/hass-aarlo/assets/16225843/fc570489-39fa-47ec-aedc-db17204e01ed">

**Comment by twrecked** [comment]
You've been moved over to locations so that one doesn't work anymore. This is an _Arlo_ thing.

Look for an entity called `alarm_control_panel.location_Home`.

**Comment by anachronox99** [comment]
yup that fixed it! Thanks. so I now appear to have 3  alarm_control_panel entities created - the second one works. I'll disable the others. 

<img width="318" alt="image" src="https://github.com/twrecked/hass-aarlo/assets/16225843/e12d87e1-9976-4922-9503-8e3590ad8a74">

**Comment by twrecked** [comment]
It's a pain for sure. I'll look to tidy this up later.  Disabling for now is the way to go.

**Comment by brenmcg57** [comment]
Looks like we're both operational.
